source/corpora.cpp

```diff
@@ -299,64 +57,28 @@ void Corpus::toJson() {
         else {
             endcomma = ",";
         }
-        std::cout << "{\"library\": \"" << library << "\", \"parent\": \"" 
+        std::cout << "{\"library\": \"" << library << "\", \"parent\": \""
                   << typeloc.parent << "\", \"name\": \"" << typeloc.name << "\", \"type\": \""
-                  << typeloc.type << "\", \"location\": \"" << typeloc.location << "\", " 
+                  << typeloc.type << "\", \"location\": \"" << typeloc.location << "\", "
                   << "\"register\": \"" << typeloc.reg << "\"}"
                   << endcomma << std::endl;
     }
-    std::cout << "]}" << std::endl; 
+    std::cout << "]}" << std::endl;
 
 }
 
 // parse a function for parameters and abi location
-void Corpus::parseFunctionABILocation(Symbol *symbol) {
-  // Get the name and type of the symbol
-  std::string sname = symbol->getMangledName();
-  std::string stype = Corpus::getStringSymbolType(symbol);
-
-  Function *func = symbol->getFunction();
-  std::vector<localVar *> params;
-
-  // The function name looks equivalent to the symbol name
-  std::string fname = func->getName();
-
-  // Get parameters with types and names
-  if (func->getParams(params)) {
-
-    // We need to keep track of the order and framebase, which starts at 8
-    std::string framebaseStr;
-    int framebase = 8;
-    int order = 1;
-
-    for (auto &param : params) {
-      std::string paramName = param->getName();
-      Type *paramType = param->getType();
-
-      // Get register class based on type
-      std::vector <RegisterClass> regClasses = getRegisterClassFromType(paramType);
-
-      // Get register name from register classes
-      std::string loc = getStringLocationFromRegisterClass(regClasses);
-
-      // This uses location lists, not reliable
-      // std::string locoffset = getParamLocationOffset(param);
-
-      // Create a new typelocation to parse later
-      TypeLocation typeloc;
-      typeloc.name = paramName;
-      typeloc.parent = fname;
-      typeloc.type = paramType->getName();
-     
-      // TODO how to determine if export/import?
-      typeloc.exportOrImport = "export";
-      typeloc.reg = loc;
-      typeloc.location = "framebase+" + std::to_string(framebase);
-      typelocs.push_back (typeloc);
-      order += 1;
-
-      // Update the framebase for the next parameter based on the type
-      framebase = updateFramebaseFromType(paramType, framebase);      
+void Corpus::parseFunctionABILocation(Dyninst::SymtabAPI::Symbol *symbol, Dyninst::Architecture arch) {
+    switch (arch) {
+      case Dyninst::Architecture::Arch_x86_64:
+        typelocs = std::move(x86_64::parse_parameters(symbol));
+        break;
+      case Dyninst::Architecture::Arch_aarch64:
```

Comment:
Haha we have such a big TODO in front of us... :)

Reply:
Oh, yeah. PPC is going to be a headache.